src/main/java/org/folio/circulation/domain/policy/LoanPolicy.java

```diff
@@ -304,11 +344,24 @@ private FixedDueDateSchedules getRenewalDueDateLimitSchedules() {
 
   private Period getRenewalPeriod(
     JsonObject loansPolicy,
-    JsonObject renewalsPolicy) {
+    JsonObject renewalsPolicy,
+    boolean isRenewalWithHoldRequest) {
 
-    return useDifferentPeriod()
-      ? getPeriod(renewalsPolicy)
-      : getPeriod(loansPolicy);
+    Period result;
+    if (isRenewalWithHoldRequest) {
+      result = getAlternateRenewalLoanPeriodForHolds();
```

Comment:
Given **alternate renewal period for holds** is not a required field, what should happen when this property is not present? 
Seems like CIRC-200 doesn't describe this case. Now it will lead to the failure if we try to renew the loan with active hold request without setting this period.

Reply:
Hi @dmytrotur, it's a good question. I guess the answer is following.

There are `renewItemsWithRequest` and `alternateRenewalLoanPeriod` properties. The first one is a boolean flag and it means whether we allow to renew an item with active pending hold request (`isRenewalWithHoldRequest` var holds this  property value).  And the second property explicitly specifies the period to use to renew **such an item** (with active pending hold requests). So, in a fact, the first property 'enables' second (`alternateRenewalLoanPeriod`, see https://github.com/folio-org/mod-circulation/pull/337/files/7cc00fbdc8be7af10bcf350c7f279208b04c968b#diff-ed7577079ed72db8b6326ac0a861757dR107).

So, if `isRenewalWithHoldRequest` is true BUT `alternateRenewalLoanPeriod` won't be specified than it is invalid case, policy configuration issue. 

I'm not sure why we have this boolean flag together with the period property as the property presence can itself indicate that such an action is allowed, but as is. 

Does it make sense?

Reply:
@bohdan-suprun Thanks for the explanation, makes sense to me. 
And this means we should add the restriction on the UI.

Reply:
@dmytrotur I guess it is out of scope, we need separate task for this.

Reply:
@bohdan-suprun Sure, it is out of the scope. Just not to forget about this, we need a story in a backlog.